app/src/main/java/com/android/streetworkapp/ui/train/TrainParamScreen.kt

```diff
@@ -0,0 +1,367 @@
+    type: String,
+    paddingValues: PaddingValues = PaddingValues(0.dp)
+) {
+  var minutes by remember { mutableIntStateOf(0) }
+  var seconds by remember { mutableIntStateOf(0) }
+  var sets by remember { mutableIntStateOf(1) }
+  var reps by remember { mutableIntStateOf(10) }
+
+  Column(
+      modifier =
+          Modifier.fillMaxSize().padding(paddingValues).padding(16.dp).testTag("TrainParamScreen"),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.Center) {
+        TrainParamHeader(
+            isTimeDependent = isTimeDependent,
+            minutes = minutes,
+            seconds = seconds,
+            sets = sets,
+            reps = reps,
+            activity = activity)
+
+        if (isTimeDependent) {
+          TimerInputGrid(
+              minutes = minutes,
+              seconds = seconds,
+              onUpdateMinutes = { minutes = it },
+              onUpdateSeconds = { seconds = it })
+        } else {
+          SetsAndRepsSection(
+              sets = sets, reps = reps, onUpdateSets = { sets = it }, onUpdateReps = { reps = it })
+        }
+
+        ConfirmActionButton(
+            navigationActions = navigationActions,
+            activity = activity,
+            isTimeDependent = isTimeDependent,
+            type = type,
+            minutes = minutes,
+            seconds = seconds,
+            sets = sets,
+            reps = reps)
+      }
+}
+
+@SuppressLint("DefaultLocale")
+@Composable
+fun TrainParamHeader(
+    isTimeDependent: Boolean,
+    minutes: Int,
+    seconds: Int,
+    sets: Int,
+    reps: Int,
+    activity: String
+) {
+  Box(
+      modifier = Modifier.fillMaxWidth().padding(vertical = 16.dp).testTag("ParamDisplay"),
+      contentAlignment = Alignment.Center) {
+        Text(
+            text =
+                if (isTimeDependent) {
+                  String.format(
+                      "I want to do %02d min and %02d seconds of %s", minutes, seconds, activity)
+                } else {
+                  String.format("I want to do %2d sets of %2d %s", sets, reps, activity)
+                },
+            style = androidx.compose.material3.MaterialTheme.typography.bodyLarge)
+      }
+}
+
+@Composable
+fun SetsAndRepsSection(
+    sets: Int,
+    reps: Int,
+    onUpdateSets: (Int) -> Unit,
+    onUpdateReps: (Int) -> Unit
+) {
+  NumberPicker(
+      label = "Number of sets:",
+      value = sets,
+      range = 0..100,
+      onValueChange = onUpdateSets,
+      modifier = Modifier.testTag("SetsPicker"))
+  NumberPicker(
+      label = "Number of reps:",
+      value = reps,
+      range = 0..100,
+      onValueChange = onUpdateReps,
+      modifier = Modifier.testTag("RepsPicker"))
+}
+
+@Composable
+fun ConfirmActionButton(
+    navigationActions: NavigationActions,
+    activity: String,
+    isTimeDependent: Boolean,
+    type: String,
+    minutes: Int,
+    seconds: Int,
+    sets: Int,
+    reps: Int
```

Comment:
I believe this approach will increase future implementation time. Instead of emphasizing argument precision, I suggest placing all arguments in the correct order. This improves readability without adding much boilerplate.
What do you think ?